src/app-layout/utils/applayout-visibility-context.tsx

```diff
@@ -0,0 +1,5 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { createContext } from 'react';
+
+export const AppLayoutVisibilityContext = createContext<boolean>(true);
```

Comment:
For compatibility with widgetization, you need to define context this way: https://github.com/cloudscape-design/components/blob/main/src/app-layout/visual-refresh-toolbar/contexts.ts#L11-L12

Also, you could keep both contexts in a single file for consistency